services/models.py

```diff
@@ -74,3 +84,14 @@ def __str__(self):
         },
         {"name": "created_at", "accessor": lambda x: x.strftime("%Y-%m-%d")},
     )
+
+    def download_gdpr_data(self):
+        """Download service specific GDPR data by profile
+        """
+        if self.service.gdpr_url:
+            url = urllib.parse.urljoin(self.service.gdpr_url, str(self.profile.pk))
+            try:
+                return requests.get(url)
```

Comment:
You should define a timeout here [since](https://requests.readthedocs.io/en/master/user/quickstart/#timeouts):
> If no timeout is specified explicitly, requests do not time out.

Reply:
To catch 4xx and 5xx errors (which in `requests` do inherit from `RequestException`) you also need to call `response.raise_for_status()`

```python
response = requests.get(url, timeout=5)
response.raise_for_status()
return response
```

Reply:
Check 👍 Thanks for the tips.